gsa/src/web/entities/__tests__/filterprovider.js

```diff
@@ -0,0 +1,257 @@
+
+import {loadingActions} from 'web/store/usersettings/defaults/actions';
+import {defaultFilterLoadingActions} from 'web/store/usersettings/defaultfilters/actions';
+import {pageFilter} from 'web/store/pages/actions';
+
+import {rendererWith, waitForElement} from 'web/utils/testing';
+
+import FilterProvider, {DEFAULT_FALLBACK_FILTER} from '../filterprovider';
+
+describe('FilterProvider component tests', () => {
+  test('should return locationQuery filter', async () => {
+    // if FilterProvider gets a locationQuery as prop, use its filter
+    const resultingFilter = Filter.fromString('location=query rows=42');
+
+    const locationQuery = {filter: 'location=query'};
+
+    const defaultSettingfilter = Filter.fromString('foo=bar');
+
+    const getSetting = jest.fn().mockResolvedValue({});
+    const gmp = {
+      user: {
+        getSetting,
+      },
+    };
+
+    const {render, store} = rendererWith({
+      gmp,
+      store: true,
+      router: true,
+    });
+
+    store.dispatch(loadingActions.success({rowsperpage: {value: '42'}}));
+    store.dispatch(
+      defaultFilterLoadingActions.success('task', defaultSettingfilter),
+    );
+
+    const renderFunc = jest.fn().mockReturnValue(<div />);
+
+    const {baseElement, getByTestId} = render(
+      <FilterProvider locationQuery={locationQuery} gmpname="task">
+        {renderFunc}
+      </FilterProvider>,
+    );
+
+    await waitForElement(() => getByTestId('awaiting-span'));
+
+    expect(renderFunc).toHaveBeenCalledWith({filter: resultingFilter});
+    expect(baseElement).toMatchSnapshot();
+  });
+
+  test('should return pageFilter', async () => {
+    // if no query is given and a pageFilter exists, use that one
+    const resultingFilter = Filter.fromString('page=filter rows=42');
+
+    const pFilter = Filter.fromString('page=filter');
+
+    const defaultSettingfilter = Filter.fromString('foo=bar');
+
+    const getSetting = jest.fn().mockResolvedValue({});
+    const gmp = {
+      user: {
+        getSetting,
+      },
+    };
+
+    const {render, store} = rendererWith({
+      gmp,
+      store: true,
+      router: true,
+    });
+
+    store.dispatch(loadingActions.success({rowsperpage: {value: '42'}}));
+    store.dispatch(
+      defaultFilterLoadingActions.success('task', defaultSettingfilter),
+    );
+    store.dispatch(pageFilter('task', pFilter));
+
+    const renderFunc = jest.fn().mockReturnValue(<div />);
+
+    const {baseElement, getByTestId} = render(
+      <FilterProvider gmpname="task">{renderFunc}</FilterProvider>,
+    );
+
+    await waitForElement(() => getByTestId('awaiting-span'));
+
+    expect(renderFunc).toHaveBeenCalledWith({filter: resultingFilter});
+    expect(baseElement).toMatchSnapshot();
+  });
+
+  test('should return defaultSettingFilter', async () => {
+    // if no pageFilter exists use the user default from the usersettings
+    const resultingFilter = Filter.fromString('foo=bar rows=42');
```

Comment:
Is it correct to merge the default rows setting into the default filter? How is it handled in gvmd currently?

Reply:
Ah I see it is only merged if rows is not set in the filter. Perfect :+1: